sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousDataSourceRDDIter.scala

```diff
@@ -177,6 +176,7 @@ class DataReaderThread(
   private[continuous] var failureReason: Throwable = _
 
   override def run(): Unit = {
+    TaskContext.setTaskContext(context)
```

Comment:
This line was added since the last PR. Why is this needed?

Reply:
TaskContext is a ThreadLocal. We need to get it in the Kafka data reader to check if an interrupt has come.

I argue that this doesn't represent an abstraction failure in the V2 API, but a workaround to the fact that Kafka consumers are badly behaved and do not respect interrupts.